File: python/domain_invitations.py
import sdk.python.entities
from sdk.python.entities import Property


class DomainInvitation(sdk.python.entities.Entity):

    resource = '/domain_invitations/'
    json_name = 'domainInvitation'

    id = Property(int)
    user_name = Property(str)
    user_email = Property(str)
    role = Property(int)


class DomainInvitations(sdk.python.entities.Resource):

    entity_class = DomainInvitation
    json_name = 'domainInvitations'


domain_invitations = DomainInvitations()

File: php/src/domain_invitations.php
<?php
declare(strict_types=1);

require_once 'entity.php';
require_once 'domains.php';
require_once 'users.php';

class DomainInvitation extends Entity
{
    public static $resource = '/domain_invitations/';
    public static $json_name = 'domainInvitation';

    public function __construct()
    {
        $this->json_property('id', 'integer');
        $this->json_property('user_name', 'string');
        $this->json_property('user_email', 'string');
        $this->json_property('role', 'integer');
        $this->json_property('domain', 'Domain', '',
                              $many = False, $recursive = True);
        $this->json_property('sender', 'User', '',
                              $many = False, $recursive = True);
        $this->json_property('user', 'User', '',
                             False, $recursive = True);
    }
}

class DomainInvitations extends Resource
{
    public function __construct()
    {
        $this->entity_class = 'DomainInvitation';
        $this->json_name = 'domainInvitations';
    }
}

$domain_invitations = new DomainInvitations();
